src/persist-client/src/shard_id.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::str::FromStr;
+
+use bytes::BufMut;
+use proptest_derive::Arbitrary;
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use mz_persist_types::Codec;
+use mz_repr::proto::newapi::{RustType, TryFromProtoError};
+
+include!(concat!(env!("OUT_DIR"), "/mz_persist_client.shard_id.rs"));
+
+/// An opaque identifier for a persist durable TVC (aka shard).
+///
+/// The [std::string::ToString::to_string] format of this may be stored durably
+/// or otherwise used as an interchange format. It can be parsed back using
+/// [str::parse] or [std::str::FromStr::from_str].
+#[derive(Arbitrary, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct ShardId([u8; 16]);
+
+impl RustType<ProtoShardId> for ShardId {
```

Comment:
counter-proposal: can we move the definition of `ProtoShardId` and this `impl RustType<ProtoShardId> for ShardId` into mz_repr? I think that would address my concerns

Reply:
if we do that, we may want to move ShardId into mz_persist_types, so that mz_repr doesn't have to depend on the entirely of the persist rich client. but I'd be fine with that move

Reply:
Hmm, that seems OK to me! Would you still be fine with keeping `ShardId` in its own module inside `persist-client`, or do you prefer for me to revert that change as well?

Reply:
Also, is it OK for `repr` to depend on `persist-client`?

Reply:
Pushed a new version that doesn't touch `persist-client` and instead offers a minimalist wrapper in `repr` around it!